pkg/proxy/ipvs/proxier.go

```diff
@@ -808,6 +808,45 @@ func (proxier *Proxier) syncProxyRules() {
 	// activeBindAddrs represents ip address successfully bind to DefaultDummyDevice in this round of sync
 	activeBindAddrs := map[string]bool{}
 
+	hasNodePort := false
+	for _, svc := range proxier.serviceMap {
+		svcInfo, ok := svc.(*serviceInfo)
+		if ok && svcInfo.NodePort() != 0 {
+			hasNodePort = true
+			break
+		}
+	}
+
+	// Both nodeAddresses and nodeIPs can be reused for all nodePort services
+	// and only need to be computed if we have at least one nodePort service.
+	var (
+		// List of node addresses to listen on if a nodePort is set.
+		nodeAddresses []string
+		// List of node IP addresses to be used as IPVS services if nodePort is
+		// set.
+		nodeIPs []net.IP
+	)
+
+	if hasNodePort {
+		nodeAddrSet, err := utilproxy.GetNodeAddresses(proxier.nodePortAddresses, proxier.networkInterfacer)
+		if err != nil {
+			klog.Errorf("Failed to get node ip address matching nodeport cidr: %v", err)
+		}
+		if nodeAddrSet.Len() > 0 {
+			nodeAddresses = nodeAddrSet.List()
+			for _, address := range nodeAddresses {
```

Comment:
For this PR, can we keep the logic the same as what it was before and open a separate PR for the zero cidr changes? 

i.e. keep this section as:
```go
                      for _, address := range nodeAddresses {
				if !utilproxy.IsZeroCIDR(address) {
					nodeIPs = append(nodeIPs, net.ParseIP(address))
					continue
				}
				// zero cidr
				nodeIPs, err = proxier.ipGetter.NodeIPs()
				if err != nil {
					klog.Errorf("Failed to list all node IPs from host, err: %v", err)
				}
			}
```

Reply:
Fwiw I think your current changes do make more sense since we should break on zero cidr but easier to isolate the PRs that way

Reply:
Not opposed to separate commits either actually

Reply:
Thanks for the feedback @andrewsykim! I have updated the PR separating the changes in 2 different commits and trying to explain the reasoning better in the commit messages.

Reply:
thank you!